File: Python/Next_Prime.py
# Modified: 07/12/2017
#
# Author: jguardado39 (John Guardado)


"""
https://www.codewars.com/kata/58e230e5e24dde0996000070

I don't think we have this specific prime kata on codewars, yet.

It's really simple:

Get the next prime number!

You will get a number n (n>=0) and your task is to find the next prime number.

e.g: nextPrime(5)=>7 || nextPrime(12)=>13

Make sure to optimize your code. There will be huge numbers in the tests!

"""

def nextPrime(n):
  return is_prime(n+1)
  
def is_prime(m):
  if m < 2: return 2
  for i in range(2, int(m ** 0.5) + 1):
    if m % i == 0:
      m += 1
      return is_prime(m)
  return m

def nextPrime_alt(n):
    if n <= 2: return 2 + (n == 2)
    p, d = n + [1, 4, 3, 2, 1, 2][n % 6], 4 - 2 * (0 < n % 6 < 5)
    while True:
        for k in range(5, int(p ** .5) + 1, 6):
            if not p % k or not p % (k + 2): break
        else: return p
        p, d = p + d, 6 - d

File: JavaScript/Next_Prime.js
// Modified: 07/12/2017
//
// Author: jguardado39 (John Guardado)

/*
  
https://www.codewars.com/kata/58e230e5e24dde0996000070

I don't think we have this specific prime kata on codewars, yet.

It's really simple:

Get the next prime number!

You will get a number n (n>=0) and your task is to find the next prime number.

e.g: nextPrime(5)=>7 || nextPrime(12)=>13

Make sure to optimize your code. There will be huge numbers in the tests!

*/

function nextPrime(n) {
  return isPrime(n + 1);
}

function isPrime(m) {
  if (m <= 2) return 2;
  for (var i = 2; i < Math.sqrt(m) + 1; i++)
    if (m % i === 0) {
      return isPrime(m + 1);
    }
   return m;
}

// By smile67
function nextPrime_alt(n){
  n = n < 1 ? 1 : n;
  while (++n) {
     for (var f = 0, i = 2; i * i <= n; i++) if (n % i == 0) {f = 1; break}
     if (!f) return n
  }
}